source/common/http/conn_manager_impl.cc

```diff
@@ -792,12 +794,14 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
   if (connection_manager_.config_.accessLogFlushInterval().has_value()) {
     access_log_flush_timer_ =
         connection_manager.read_callbacks_->connection().dispatcher().createTimer([this]() -> void {
-          // If the request is complete, we've already done the stream-end access-log, and shouldn't
-          // do the periodic log.
-          if (!streamInfo().requestComplete().has_value()) {
-            filter_manager_.log();
-            refreshAccessLogFlushTimer();
+          // If the request is not in active state, we shouldn't do the periodic log.
+          if (!streamInfo().streamState() ||
+              streamInfo().streamState() != StreamInfo::StreamState::InProgress) {
+            return;
```

Comment:
This should schedule another callback, no? If it is "Started" it'll never call again.

Reply:
Yeah you're right, modified it, thanks